ssoPlugin/widget/index.html

```diff
@@ -0,0 +1,278 @@
+		                      value and the API client object, rejects with the API error response.
+		 */
+		const checkAuthenticationStatus = (authApiSettings) => new Promise((resolve, reject) => {
+			createAuth0Client({
+				domain: authApiSettings.domain,
+				client_id: authApiSettings.clientId,
+				// cacheLocation: 'localstorage',
+			})
+				.then(auth0 => {
+					auth0.isAuthenticated()
+						.then((isAuthenticated) => resolve({ isAuthenticated, auth0 }))
+						.catch((error) => reject(error))
+				})
+				.catch((error) => reject(error))
+		})
+
+		/**
+		 * Provided an Auth0 API client, gets the active user object and stores it in localStorage.
+		 *
+		 * @returns {Promise} Resolves true, or rejects with an error message.
+		 */
+		const getAndStoreUser = (auth0) => new Promise((resolve, reject) => {
+			auth0.getUser()
+				.then((user) => {
+					saveToLocalStorage({ user })
+						.then(() => resolve(true))
+						.catch((error) => reject('There was a problem storing your authentication data.'))
+				})
+				.catch((error) => reject(error))
+		})
+
+		/**
+		 * Handles a confirmed authentication by trying to get/store the user object and then
+		 * navigating to the RSS plugin.
+		 */
+		const handleAuthenticated = (auth0) => {
+			$body.classList.remove('unauthenticated')
+
+			getAndStoreUser(auth0)
+				.then(() => {
+					$body.classList.add('authenticated')
+					goToRss()
+				})
+				.catch((error) => alertMessage(`${error} Please try again.`, 'error'))
+				.finally(() => initComplete())
+		}
+
+		/**
+		 * Handles a confirmed lack of authentication by cleaning up the DOM and binding login actions
+		 * to the login button.
+		 */
+		const handleUnauthenticated = (auth0) => {
+			initComplete()
+			$body.classList.add('unauthenticated')
+
+			$loginButton.addEventListener('click', (e) => {
+				e.preventDefault()
+
+				$loginButton.disabled = true
+
+				auth0.loginWithPopup()
+					.then(() => handleAuthenticated(auth0))
+					.catch((error) => alertMessage('There was a problem trying to log you in.', error))
+					.finally(() => $loginButton.disabled = false)
+			})
+		}
+
+		/**
+		 * Navigates the user to the RSS Plugin.
+		 *
+		 * This currently does not work because the BuildFire SDK throws an inscrutable error, so
+		 * temporarily just telling the user to navigate to the user. Will fix.
+		 */
+		const goToRss = () => {
+			// TODO: Figure out:
+			//       (1) why this doesn't seem to work in the SDK, and
+			//       (2) how we can navigate reliably to the RSS Plugin.
+			// buildfire.navigation.navigateTo({
+			// 	pluginId: 'rssPlugin',
+			// 	instanceId: '1591887442595-005759697297339572',
+			// 	title: 'RSS Plugin',
+			// })
+			// Thus, this very temporary workaround:
+			const tempMessage = document.createTextNode("Please navigate to the RSS Plugin. (The BuildFire SDK environment doesn't support navigating there automatically.)")
```

Comment:
We'll want to replace this text with something more user friendly, even in the short term.

E.g. "You're now logged in! Please carry on." lolidk

Reply:
I can't even tell you what simmering rage underlies the message as-is. How, _how_ does the BuildFire SDK not allow navigating from one plugin to another? 😡

Reply:
In any case: sure, I'll change the messaging. There is likely going to be a stage (like, now) where we're testing this and will have to just _know_ that "Please carry on" means "now navigate to the RSS plugin manually", which I don't love. But I hope to have automatic navigation working fast enough it doesn't matter what the messaging is, so, I happily defer and will change!